### PR TITLE
Cleanups, fixes and a bit of optimizations for site/components batch #4

### PR DESCRIPTION
**Pull request by frankmayer**
- com_mailto
- com_newsfeeds
- com_search

Note: This is a single commit bundling all types of changes, since PR #12261 which had detailed commits, was rejected as a whole

### CODE REVIEW

File: components/com_mailto/controller.php [hunk 1]
```diff
@@ -135,7 +135,6 @@ public function send()
 
 		// Build the message to send
 		$msg  = JText::_('COM_MAILTO_EMAIL_MSG');
-		$link = $link;
```
Comment (andrepereiradasilva) [inline components/com_mailto/controller.php]: nice one! :wink:

File: components/com_newsfeeds/views/newsfeed/view.html.php [hunk 2]
```diff
@@ -139,7 +139,7 @@ public function display($tpl = null)
 		// Check the access to the newsfeed
 		$levels = $user->getAuthorisedViewLevels();
 
-		if (!in_array($item->access, $levels) or ((in_array($item->access, $levels) and (!in_array($item->category_access, $levels)))))
+		if (!in_array($item->access, $levels) or in_array($item->access, $levels) and (!in_array($item->category_access, $levels)))
```
Comment (andrepereiradasilva) [inline components/com_newsfeeds/views/newsfeed/view.html.php]: why not replace the `and` and `or` here?

Comment (frankmayer) [inline components/com_newsfeeds/views/newsfeed/view.html.php]: Skipped through...

Comment (andrepereiradasilva) [inline components/com_newsfeeds/views/newsfeed/view.html.php]: also maybe you need to add one parenthisis here. not sure tough

Comment (frankmayer) [inline components/com_newsfeeds/views/newsfeed/view.html.php]: Yes, precedence maybe.. I mostly picked low hanging fruit..  That's possibly why this one "slipped through" :smile:

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on c34e7bfa4c3bdba9672de58c8ac495c79b618ec6<br /><br />on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12293">issues.joomla.org/tracker/joomla-cms/12293</a>.</sub>

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 2b4d825c643e85d7d38e12cb64c022a1a71c5c9c<br /><br />code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12293">issues.joomla.org/tracker/joomla-cms/12293</a>.</sub>

**Comment by frankmayer** [comment]
Fixed some conflicts that had occurred in the meantime.
Having merged the changes of above referenced PR's made this PR lighter and easier to test.
It is essentially ready to be merged, when the team finds the time and PR is OK.

**Review by andrepereiradasilva — APPROVED** [review]

File: components/com_newsfeeds/controller.php [hunk 3]
```diff
@@ -36,7 +36,7 @@ public function display($cachable = false, $urlparams = false)
 
 		$user = JFactory::getUser();
 
-		if ($user->get('id') || ($this->input->getMethod() == 'POST' && $vName == 'category' ))
+		if ($user->get('id') || ($this->input->getMethod() === 'POST' && $vName === 'category' ))
```
Comment (Quy) [inline components/com_newsfeeds/controller.php]: Remove extra space `'category' ))`.

File: components/com_newsfeeds/models/newsfeed.php [hunk 4]
```diff
@@ -127,7 +127,7 @@ public function &getItem($pk = null)
 
 				// Check for published state if filter set.
 
-				if (((is_numeric($published)) || (is_numeric($archived))) && (($data->published != $published) && ($data->published != $archived)))
+				if ((is_numeric($published) || is_numeric($archived)) && (($data->published != $published) && ($data->published != $archived)))
```
Comment (Quy) [inline components/com_newsfeeds/models/newsfeed.php]: Change to `$data->published != $published && $data->published != $archived`.

File: components/com_newsfeeds/router.php [hunk 5]
```diff
@@ -58,20 +58,18 @@ public function build(&$query)
 		}
 
 		// Are we dealing with a newsfeed that is attached to a menu item?
-		if (isset($query['view']) && ($mView == $query['view']) and (isset($query['id'])) and ($mId == (int) $query['id']))
+		if (isset($query['view'], $query['id']) && ($mView == $query['view']) && ($mId == (int) $query['id']))
```
Comment (Quy) [inline components/com_newsfeeds/router.php]: Change `&& ($mView == $query['view']) && ($mId == (int) $query['id'])` to `&& $mView == $query['view'] && $mId == (int) $query['id'].

File: components/com_newsfeeds/views/category/tmpl/default_items.php [hunk 6]
```diff
@@ -20,9 +20,9 @@
 <?php else : ?>
 
 <form action="<?php echo htmlspecialchars(JUri::getInstance()->toString(), ENT_COMPAT, 'UTF-8'); ?>" method="post" name="adminForm" id="adminForm">
-	<?php if ((($this->params->get('filter_field') != 'hide') || ($this->params->get('filter_field') != '0')) || $this->params->get('show_pagination_limit')) :?>
+	<?php if ((($this->params->get('filter_field') !== 'hide') || ($this->params->get('filter_field') != '0')) || $this->params->get('show_pagination_limit')) :?>
```
Comment (Quy) [inline components/com_newsfeeds/views/category/tmpl/default_items.php]: Remove parentheses.

File: components/com_newsfeeds/views/category/tmpl/default_items.php [hunk 7]
```diff
@@ -20,9 +20,9 @@
...
 	<fieldset class="filters btn-toolbar">
-		<?php if (($this->params->get('filter_field') != 'hide') || ($this->params->get('filter_field') != '0')) :?>
+		<?php if (($this->params->get('filter_field') !== 'hide') || ($this->params->get('filter_field') != '0')) :?>
```
Comment (Quy) [inline components/com_newsfeeds/views/category/tmpl/default_items.php]: Remove parentheses

File: components/com_newsfeeds/views/newsfeed/tmpl/default.php [hunk 8]
```diff
@@ -17,31 +17,33 @@
 {
 	$lang      = JFactory::getLanguage();
 	$myrtl     = $this->newsfeed->rtl;
-	$direction = " ";
+	$direction = ' ';
 
-		if ($lang->isRtl() && $myrtl == 0)
+	$isRtl= $lang->isRtl();
```
Comment (Quy) [inline components/com_newsfeeds/views/newsfeed/tmpl/default.php]: Align `=`

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 6cc4c65e078da439a031551e6e68778636aa35dc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12293">issues.joomla.org/tracker/joomla-cms/12293</a>.</sub>

File: components/com_newsfeeds/router.php [hunk 9]
```diff
@@ -147,14 +147,14 @@ public function getCategoryId($segment, $query)
...
-					if ($child->alias == $segment)
+					if ($child->alias === $segment)
 					{
 						return $child->id;
 					}
 				}
 				else
 				{
-					if ($child->id == (int) $segment)
+					if ($child->id === (int) $segment)
```
Comment (andrepereiradasilva) [inline components/com_newsfeeds/router.php]: are you sure the categories child id is a integer?
see https://github.com/joomla/joomla-cms/blob/staging/libraries/legacy/categories/categories.php#L722 for instance

**Review by andrepereiradasilva — COMMENTED** [review]
one question with strict compare a integer type with mixed(?) type. please check
the rest seems fine

File: components/com_newsfeeds/router.php (hunk 9 above)
Comment (frankmayer) [inline components/com_newsfeeds/router.php]: Not sure. Will check. According to the property's docblock it is an integer, but I am not sure if we can trust that. See: https://github.com/joomla/joomla-cms/blob/staging/libraries/legacy/categories/categories.php#L395

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 5b883303a79388f35cb8443a123a2230366859d5<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12293">issues.joomla.org/tracker/joomla-cms/12293</a>.</sub>

**Comment by frankmayer** [comment]
@franz-wohlkoenig RTC please , thank you

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.

**Pull request merged**